blog/ca/5_descubrint-el-ecommerce-de-nike.mdx

```diff
@@ -0,0 +1,1125 @@
+showReviews:true
+showSEOCopy:true
+showSinglesDaySalesLink:false
+showSizeAndFit:true
+showSizeCalculator:false
+showSocialLinksNBY:true
+showUniteTimers:false
+showWidthSelector:true
+showWishlist:true
+skipPercentOff:false
+skipSegmentDupedEvents:true
+skipSustainability:false
+swooshEligibleGeo:false
+useBounceExchange:false
+useDotcomNav:true
+useDotcomNavESI:true
+useExclusiveAccess:true
+useFeedsMoreInfoMessage:true
+useFeedsSizeFitMessage:true
+useFeedsSustainabilityMessage:true
+useMAPCollection:true
+useNBYCloudUrl:true
+useNBYImageService:true
+useNeo:true
+useNewRelic:true
+useNotifyMeV2:false
+useOfferRFY:false
+useOptimizelyX:false
+usePersonalizedNullSearch:false
+useRollupKey:true
+useSentryIO:false
+useServiceCanonicalUrl:true
+useSwitchboardRecommendations:false
+useTTReviews:true
+useTTV5Reviews:true
+useUniteStaging:false
+wallFavorites:false
+wallFiltersBtnFilterLabel:false
+wallImpressions:true
+wallNoResultsCarousel:true
+```
+
+### Observabilitat
+
+Si mirem la consola del navegador podem veure que nike esta utiltizant el servi de [newrelic](https://newrelic.com/) per capturar expecions i errors del client:
+
+```js:_N_E/[[...page]].js
+const isAdobeTargetEnabled = process.env.ADOBE_TARGET_ENABLED === 'true';
+
+export const getServerSideProps = wrapper.getServerSideProps(
+  store =>
+    async ({ req, query, res }) => {
+      const lpContext = {
+        store,
+        query,
+      };
+
+      queryParamsSanitizerMiddleware(req);
+
+      // error handling should be initiated before the middlewares called as they can dispatch error actions
+      store.dispatch(actions.logActions.init({ logger: req.logger, newrelic }));
+      // call our old express middlewares
+      featureFlagsMiddleware(req, lpContext);
+
+      lpContext.fetchFn = getExtendedFetchFn(store.getState());
+      // TODO: CICLP-3564 Required for data fetching in server side sagas. To be removed during cleanup
+      globalThis.extendedFetch = lpContext.fetchFn;
+```
+
+## Graphql
+
+Una de les preguntes que em feia era si feien servir graphql o tot eran api custom/restful, tot i que clarament dominen més les apis rest, si que utiltizen graphql, en aquest cas especificament pel tema de les wishlists a l'hora de recuperar els items per exemple.
+
+```js
+import graphqlFetch from '@nike/graphql-fetch';
+import { fetchClient } from '@nike/fetch-client';
+import { DOTCOM_UX_ID } from '@nike/ux-tread-dotcom-constants';
+
+const graphqlClient = (args) =>
+  graphqlFetch({
+    fetchClient,
+    url: process.env.GRAND_URL || 'https://api.nike.com/cic/grand/v1/graphql',
+    callerId: window.NikeShop?.getVersion(),
+    appId: DOTCOM_UX_ID,
+    ...args,
+  });
+
+export default graphqlClient;
+```
+
+A part d'això no n'he pogut extreure gaire res més ja que fent la tipica petició per recuperar informació sobre el schema no retorna res (429 too many request).
+
+![react-devtools](/assets/blog/descubrint-el-ecommerce-de-nike/graphql-error.PNG)
```

Comment:
Afegiria una mini secció de concluió/resum molt sencill i esquemàtic, per tenir un overview del reverse enginering de la web de nike. Podria ser algo com:

* Framework: Utilitzen Next.js, com es pot veure pel div amb l'id __next.
* State Management: Redux
* CDN: Fan servir Akamai
* Sistema de Disseny: Utilitzen Emotion (CSS-in-JS) i un sistema de disseny prop
* Internacionalització: Gestionen les traduccions i el i18n a través de fitxers JSON específics per idioma.
* Analítiques: Empren serveis com Optimizely i Adobe Target per a l'anàlisi i la segmentació dels usuaris.
* Feature Flags: Utilitzen un extens llistat de feature flags per gestionar funcionalitats específiques de la web.
* Observabilitat: Emprant New Relic per a la monitorització i l'anàlisi d'errors.